packages/hardhat-plugin/src/ui/pretty-event-handler.ts

```diff
@@ -261,16 +261,23 @@ export class PrettyEventHandler implements ExecutionEventListener {
       batches: this._applyResultToBatches(this.state.batches, event.result),
     };
 
+    const anythingDone = this.state.currentBatch > 0;
+
     // If batches where executed, rerender the last batch
-    if (this.state.currentBatch > 0) {
+    if (anythingDone) {
       this._redisplayCurrentBatch();
```

Comment:
Lets pull out `this.state.currentBatch > 0` as a util function, and use it here but also use it in `calculateDeploymentCompleteDisplay`.

Reply:
I have pushed a commit that takes the util function route. It means that `calculateDeploymentCompleteDisplay` can keep the interface of accepting `UiState` and then deriving what to display, so better encapsulation.
Looking at what had to be done to get there though makes me feel it might not have been worth it.